sherpa/include/sherpa/myArray.hh

```diff
@@ -19,567 +16,538 @@
-    }
+  T *operator()() const { return &vec[0]; }
 
-    inline Real& operator () ( int arg ) { return this->operator[]( arg ); }
-    
-    Array1D< Real >& operator += ( const Array1D< Real >& rhs ) {
-      my_cmp_sizes( rhs );
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) += rhs[ ii ];
-      return *this;
-    }
+  operator T *() { return &vec[0]; }
 
-    Array1D< Real >& operator -= ( const Array1D< Real >& rhs ) {
-      my_cmp_sizes( rhs );
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) -= rhs[ ii ];
-      return *this;
-    }
+  T &operator[](int arg) {
+#ifndef NDEBUG
+    return vec.at(arg);
+#endif
+    return vec[arg];
+  }
+  const T &operator[](int arg) const {
+#ifndef NDEBUG
+    return vec.at(arg);
+#endif
+    return vec[arg];
+  }
 
-    Array1D< Real >& operator *= ( const Array1D< Real >& rhs ) {
-      my_cmp_sizes( rhs );
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) *= rhs[ ii ];
-      return *this;
-    }
+  // std::vector<T>::iterator begin() { return vec.begin(); }//
+  // std::vector<T>::const_iterator begin() const { return vec.begin(); }
+  void resize(std::vector<int>::size_type n) { vec.resize(n); }
 
-    Array1D< Real >& operator /= ( const Array1D< Real >& rhs ) {
-      my_cmp_sizes( rhs );
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) /= rhs[ ii ];
-      return *this;
-    }
+  std::vector<int>::size_type size() const { return vec.size(); }
 
-    Array1D< Real >& operator += ( Real rhs ) {
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) += rhs;
-      return *this;
-    }
+  const std::vector<T> &get_vec() const { return vec; }
 
-    Array1D< Real >& operator -= ( Real rhs ) {
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) -= rhs;
-      return *this;
-    }
-
-    Array1D< Real >& operator *= ( Real rhs ) {
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) *= rhs;
-      return *this;
-    }
+  virtual std::ostream &print(std::ostream &s) const {
+    for (int ii = 0; ii < size(); ++ii)
+      s << vec[ii] << ' ';
+    return s;
+  }
 
-    Array1D< Real >& operator /= ( Real rhs ) {
-      for ( size_t ii = 0; ii < rhs.size( ); ++ii )
-	this->operator[]( ii ) /= rhs;
-      return *this;
-    }
+  virtual void sort() { std::sort(vec.begin(), vec.end()); }
 
-    std::ostream& myprint( std::ostream& os, int num=4, const char* prefix=NULL,
-			   const char* suffix=NULL ) const {
+  Array1D<T> &operator+=(const Array1D<T> &rhs) {
+    MYCMPSIZES(rhs);
+    for (int ii = 0; ii < rhs.size(); ++ii)
+      vec[ii] += rhs[ii];
+    return *this;
+  }
 
-      if ( this->size( ) > 0 ) {
-	if ( NULL != prefix )
-	  os << prefix << "[0]=";
-	os << this->operator[]( 0 );
-      }
+  Array1D<T> &operator-=(const Array1D<T> &rhs) {
+    MYCMPSIZES(rhs);
+    for (int ii = 0; ii < rhs.size(); ++ii)
```

Comment:
```suggestion
    for (std::vector<int>::size_type ii = 0; ii < rhs.size(); ++ii)
```

Reply:
I ask as a "not a C++ expert": should we be looking at re-writing these using `std::transform` (I think we can now assume we have this, but I can't say for sure), or is it okay to stick with the more-direct iteration here? The strong advantage in not changing the code is that there's less chance to introduce errors, but do we want to move to a more STL-like interface?